src/handlers/psp.ts

```diff
@@ -249,68 +250,48 @@ const initializeProviders = async (state: State): Promise<State> => {
-    const [transactionCountLogs, walletData] = await getSponsorWalletAndTransactionCount(
-      airnodeWallet,
-      provider,
-      currentBlock,
-      sponsorAddress
-    );
-
-    // Skip processing for the current sponsorAddress if the wallet functions fail
-    if (isNil(walletData)) {
-      const sponsorLogOptions = buildLogOptions('additional', { sponsor: sponsorAddress }, providerLogOptions);
-      utils.logger.warn('Failed to fetch sponsor wallet or transaction count', sponsorLogOptions);
-      return;
-    }
-
-    const { sponsorWallet, transactionCount } = walletData;
-
-    const sponsorWalletLogOptions = buildLogOptions(
-      'additional',
-      { sponsorWallet: shortenAddress(sponsorWallet.address) },
-      providerLogOptions
-    );
-    utils.logger.logPending(transactionCountLogs, sponsorWalletLogOptions);
-
-    utils.logger.info(`Processing ${subscriptions.length} subscription(s)`, sponsorWalletLogOptions);
-
-    const logs = await processSponsorWallet(
-      airnodeWallet,
-      contracts['DapiServer'],
-      gasTarget,
-      subscriptions,
-      sponsorWallet,
-      voidSigner,
-      transactionCount
-    );
-
-    logs.forEach(([logs, data]) => {
-      const subscriptionLogOptions = buildLogOptions(
-        'additional',
-        { subscriptionId: data.id },
-        sponsorWalletLogOptions
-      );
-      utils.logger.logPending(logs, subscriptionLogOptions);
-    });
-  });
-
-  await Promise.all(providerSponsorPromises);
+export const spawn = ({
+  providerSponsorSubscription,
+  baseLogOptions,
+  type,
+}: {
+  providerSponsorSubscription: ProviderSponsorSubscriptions;
+  baseLogOptions: utils.LogOptions;
+  type: 'local' | 'aws' | 'gcp';
+}) => {
+  // lambda.invoke is synchronous so we need to wrap this in a promise
+  switch (type) {
+    case 'local':
+      return processSubscriptionsHandler({ body: JSON.stringify({ providerSponsorSubscription, baseLogOptions }) });
+    case 'aws':
+      return new Promise((resolve, reject) => {
+        // Uses the current region by default
+        const lambda = new AWS.Lambda();
+
+        // TODO: do we want to use the airnode format for FunctionName
+        // AWS doesn't allow uppercase letters in lambda function names
+        // const resolvedName = `airkeeper-${providerSponsorSubscription.sponsorAddress}-run`;
+
+        const options = {
+          FunctionName: 'process-subscriptions',
+          Payload: JSON.stringify({ providerSponsorSubscription, baseLogOptions }),
+          // TODO: could use InvocationType: 'Event' if we don't need to wait for spawned lambdas to finish
+          // InvocationType: 'Event',
+        };
+        lambda.invoke(options, (err, data) => {
+          console.log('spawn', err, data);
+          if (err) {
+            reject(err);
+            return;
+          }
+          resolve(JSON.parse(JSON.parse(data.Payload as string).body));
+        });
+      });
+  }
```

Comment:
The `spawn` function is basically copied from Airnode, but I was wondering if you saw a point to use the `InvocationType: 'Event'` field if we don't need to keep the parent lambda running until all processing finishes.

Reply:
I don't think we should do that. I'd actually say that we want to make sure that the lambdas are done within a certain time limit. We still need to be done with the main lambda invocation within 60 seconds so the next run can start. And I think we're doing something like that on Airnode as well. I think the spawned lambdas are wrapped in a timeout.